features/machine/machine.feature

```diff
@@ -377,22 +376,39 @@ Feature: Machine features testing
-      | ["spec"]["template"]["spec"]["providerSpec"]["value"]["memoryMiB"]                        | <%= cb.memoryMiB %>                             |
-      | ["spec"]["template"]["spec"]["providerSpec"]["value"]["template"]                         | <%= cb.template %>                              |
-      | ["spec"]["template"]["metadata"]["labels"]["machine.openshift.io/cluster-api-machineset"] | default-valued-33380                            |
+      | n                                                                                         | openshift-machine-api                       |
+      | ["spec"]["selector"]["matchLabels"]["machine.openshift.io/cluster-api-cluster"]           | <%= infrastructure("cluster").infra_name %> |
+      | ["spec"]["selector"]["matchLabels"]["machine.openshift.io/cluster-api-machineset"]        | <name>                                      |
+      | ["spec"]["template"]["metadata"]["labels"]["machine.openshift.io/cluster-api-cluster"]    | <%= infrastructure("cluster").infra_name %> |
+      | ["spec"]["template"]["spec"]["providerSpec"]["value"]["workspace"]["datacenter"]          | <%= cb.datacenter %>                        |
+      | ["spec"]["template"]["spec"]["providerSpec"]["value"]["workspace"]["datastore"]           | <%= cb.datastore %>                         |
+      | ["spec"]["template"]["spec"]["providerSpec"]["value"]["workspace"]["folder"]              | <%= cb.folder %>                            |
+      | ["spec"]["template"]["spec"]["providerSpec"]["value"]["workspace"]["resourcePool"]        | <%= cb.resourcePool %>                      |
+      | ["spec"]["template"]["spec"]["providerSpec"]["value"]["workspace"]["server"]              | <%= cb.server %>                            |
+      | ["spec"]["template"]["spec"]["providerSpec"]["value"]["diskGiB"]                          | <diskGiB>                                   |
+      | ["spec"]["template"]["spec"]["providerSpec"]["value"]["memoryMiB"]                        | <%= cb.memoryMiB %>                         |
+      | ["spec"]["template"]["spec"]["providerSpec"]["value"]["template"]                         | <template>                                  |
+      | ["spec"]["template"]["metadata"]["labels"]["machine.openshift.io/cluster-api-machineset"] | <name>                                      |
+      | ["metadata"]["name"]                                                                      | <name>                                      |
+    Then the step should succeed
+
+    And I run the :get background admin command with:
+      | resource      | machine           |
+      | w             | true              |
     Then the step should succeed
-     
+
+    And 120 seconds have passed
+    When I terminate last background process
+    Then the output should contain:
+      | Provisioned  |
```

Comment:
Perhaps we can use below steps to replace line 399 ~ line 407, which should be more robust, and a little faster as we do not need to hard wait for 120 seconds.
```
  And I wait up to 120 seconds for the steps to pass:
  """
  When I run the :get admin command with:
    | resource | machine |
    | w        | true    |
  Then the step should succeed
  And the output should contain:
    | Provisioned  |
  """
```

Reply:
The steps get hung if I use above , even when Provisioned is obtained , hence I used the hard coded steps .. 
          selfLink: ""
    
    [03:39:53] INFO> Exit Status: 0
    [03:39:53] INFO> Shell Commands: rm -r -f -- /root/workdir/febf767459e0-
    
    [03:39:53] INFO> Exit Status: 0
    [03:39:53] INFO> === End After Scenario: Implement defaulting machineset values for vsphere, Examples (#1) ===

1 scenario (1 passed)
18 steps (18 passed)
19m19.859s
[03:39:53] INFO> === At Exit ===


When used hard coded --
1 scenario (1 undefined)
22 steps (3 skipped, 1 undefined, 18 passed)
2m14.998s

You can implement step definitions for undefined steps with these snippets:

Then(/^the step should succeed\t    Then the step should succeed$/) do
  pending # Write code here that turns the phrase above into concrete actions
end

[03:44:07] INFO> === At Exit ===

Let me know if what we can review for this , if the way steps are used is not correct or something else ..

Reply:
Try this,
```
  And I wait up to 120 seconds for the steps to pass:
  """
  When I run the :get admin command with:
    | resource | machine |
  Then the step should succeed
  And the output should contain:
    | Provisioned  |
  """
```

Removed the line `| w        | true    |`, as we do not need to watch in this way.

Reply:
thanks , I works now and saves time